pupa/cli/commands/clean.py

```diff
@@ -0,0 +1,103 @@
+import itertools
+from datetime import datetime, timezone, timedelta
+
+import django
+from django.apps import apps
+from .base import BaseCommand
+
+
+def get_subclasses(app_list, abstract_class):
+    """
+    Finds and returns all subclasses of an abstract class.
+    """
+    result = []
+    for app in app_list:
+        for model in apps.get_app_config(app).get_models():
+            if issubclass(model, abstract_class) and model is not abstract_class:
+                result.append(model)
+        return result
+
+
+def get_stale_objects(window):
+    """
+    Find all database objects that haven't seen been in {window} days.
+    """
+
+    from opencivicdata.core.models.base import OCDBase
+
+    ocd_apps = ["core", "legislative"]
+    # Check all subclasses of OCDBase
+    models = get_subclasses(ocd_apps, OCDBase)
+
+    results = []
+    for model in models:
+        # Jurisdictions are protected from deletion
+        if "Jurisdiction" not in model.__name__:
+            cutoff_date = datetime.now(tz=timezone.utc) - timedelta(days=window)
+            results.append(model.objects.filter(last_seen__lte=cutoff_date))
+
+    return itertools.chain(*results)
```

Comment:
What do you think about making this a generator (yielding from results) instead of building up a list?